tests/mu_pri_enum_test.c

```diff
@@ -0,0 +1,152 @@
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'B');
+    POOR_MANS_ASSERT(valueAsString[1] == 'C');
+    POOR_MANS_ASSERT(valueAsString[2] == '\0');
+    POOR_MANS_ASSERT(valueAsInt == BC);
+    return 0;
+}
+
+static int verifyMU_ENUM_VALUE_BC_plus_1(const char* empty, const char* valueAsString, int valueAsInt)
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'N');
+    POOR_MANS_ASSERT(valueAsString[1] == 'U');
+    POOR_MANS_ASSERT(valueAsString[2] == 'L');
+    POOR_MANS_ASSERT(valueAsString[3] == 'L');
+    POOR_MANS_ASSERT(valueAsString[4] == '\0');
+    POOR_MANS_ASSERT(valueAsInt == (int)BC + 1);
+    return 0;
+}
+
+#define TEST_PRI_ENUM_2_VALUES    \
+    X2, 2,                        \
+    ZY2, 6
+
+MU_DEFINE_ENUM_2(TEST_PRI_ENUM_2, TEST_PRI_ENUM_2_VALUES);
+MU_DEFINE_ENUM_STRINGS_2(TEST_PRI_ENUM_2, TEST_PRI_ENUM_2_VALUES);
+
+static int verifyMU_ENUM_VALUE_X2(const char* empty, const char* valueAsString, int valueAsInt)
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'X');
+    POOR_MANS_ASSERT(valueAsString[1] == '2');
+    POOR_MANS_ASSERT(valueAsString[2] == '\0');
+    POOR_MANS_ASSERT(valueAsInt == X2);
+    return 0;
+}
+
+static int verifyMU_ENUM_VALUE_ZY2(const char* empty, const char* valueAsString, int valueAsInt)
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'Z');
+    POOR_MANS_ASSERT(valueAsString[1] == 'Y');
+    POOR_MANS_ASSERT(valueAsString[2] == '2');
+    POOR_MANS_ASSERT(valueAsString[3] == '\0');
+    POOR_MANS_ASSERT(valueAsInt == ZY2);
+    return 0;
+}
+
+static int verifyMU_ENUM_VALUE_1(const char* empty, const char* valueAsString, int valueAsInt)
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'N');
+    POOR_MANS_ASSERT(valueAsString[1] == 'U');
+    POOR_MANS_ASSERT(valueAsString[2] == 'L');
+    POOR_MANS_ASSERT(valueAsString[3] == 'L');
+    POOR_MANS_ASSERT(valueAsString[4] == '\0');
+    POOR_MANS_ASSERT(valueAsInt == 1);
+    return 0;
+}
+
+static int verifyMU_ENUM_VALUE_3(const char* empty, const char* valueAsString, int valueAsInt)
+{
+    POOR_MANS_ASSERT(empty != NULL);
+    POOR_MANS_ASSERT(empty[0] == '\0');
+    POOR_MANS_ASSERT(valueAsString != NULL);
+    POOR_MANS_ASSERT(valueAsString[0] == 'N');
+    POOR_MANS_ASSERT(valueAsString[1] == 'U');
+    POOR_MANS_ASSERT(valueAsString[2] == 'L');
+    POOR_MANS_ASSERT(valueAsString[3] == 'L');
+    POOR_MANS_ASSERT(valueAsString[4] == '\0');
```

Comment:
This is memcmp

Reply:
In the other places too.

---
In reply to: [335810296](https://github.com/Azure/azure-macro-utils-c/pull/34#discussion_r335810296) [](ancestors = 335810296)

Reply:
Changed to use memcmp.

---
In reply to: [335810314](https://github.com/Azure/azure-macro-utils-c/pull/34#discussion_r335810314) [](ancestors = 335810314,335810296)